feat: Test for Prisma Client auto-install on generate #19259

Templer44 opened the pull request:
Tests created for issues #19259 and #19814 to check for [object promise] in error message on generate when using yarn in a Windows environment.

Also implemented code to describe [object Promise] if displayed in error message.

Templer44 committed:
fix: Fix error messages stating install with [object Promise] #19259

Co-authored-by: Sunveer Bhullar <sunveerbhullar@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prisma/prisma?pullRequest=20604) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prisma/prisma?pullRequest=20604) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prisma/prisma?pullRequest=20604) it.</sub>

Jolg42 commented on packages/client/tests/e2e/prisma-client-error-message-win32/readme.md:
````diff
@@ -0,0 +1,8 @@
+#Prisma Client Test for Windows Environment
+
+```
+Tests were created to check if the prisma client is installed properly for the Windows environment.
+For the Windows environments, the prisma client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the @prisma/client or resolve the path to or packages for the @prisma/client.
+
+Thus, Windows users must manually check if both the packages and @prisma/client are installed properly in order to successfully run the client.
+```
````
```suggestion
# Prisma Client Test on Windows

```
Tests were created to check if Prisma Client is installed properly on Windows.
On Windows, Prisma Client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the `@prisma/client` or resolve the path to or packages for `@prisma/client`.

Thus, Windows users must manually check if both the packages and `@prisma/client` are installed properly in order to successfully run the client.
```
```

Jolg42 commented on packages/client/tests/e2e/prisma-client-error-message-win32/readme.md:
````diff
@@ -0,0 +1,8 @@
+#Prisma Client Test for Windows Environment
+
+```
+Tests were created to check if the prisma client is installed properly for the Windows environment.
+For the Windows environments, the prisma client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the @prisma/client or resolve the path to or packages for the @prisma/client.
+
+Thus, Windows users must manually check if both the packages and @prisma/client are installed properly in order to successfully run the client.
````
```suggestion
Tests were created to check if Prisma Client is installed properly on Windows.
On Windows, Prisma Client is unable to resolve packages and paths automatically compared to other environments when using Yarn. This results in a "Could not resolve @prisma/client..." error. Meaning that the client is unable to find the `@prisma/client` or resolve the path to or packages for `@prisma/client`.

Thus, Windows users must manually check if both the packages and `@prisma/client` are installed properly in order to successfully run the client.
```
Small changes

Jolg42 commented on packages/client/tests/e2e/prisma-client-error-message-win32/readme.md:
```diff
@@ -0,0 +1,8 @@
+#Prisma Client Test for Windows Environment
```
```suggestion
# Prisma Client Test on Windows
```

Jolg42 commented:
(I still need to review the test)

Templer44 committed:
feat(update): Updated test files and prismaClientResolver

Co-authored-by: Sunveer Bhullar <sunveerbhullar@gmail.com>

Templer44 committed:
Updated Readme from Suggestions made.

Co-authored-by: Joël Galeran <Jolg42@users.noreply.github.com>

Templer44 commented:
Updated the tests to show all tests passing and removed comments from the catchObjPromise test file. I've also removed the changes made to the prismaClientResolver file.

Jolg42 committed:
Merge branch 'main' into prismaClientErrorMessage

jkomyno committed:
Merge branch 'main' into prismaClientErrorMessage

jkomyno commented:
Hi @Templer44, I know it's been a while, but may I please ask you to update your PR to fix linting tests? (See [CI logs](https://github.com/prisma/prisma/actions/runs/10040748519/job/27752982550?pr=20604)).

Thanks!

jkomyno commented:
Closing due to inactivity.
